Src/LogData.c

```diff
@@ -36,316 +73,209 @@ void writeToEEPROM(uint8_t* buffer, uint16_t bufferSize, uint16_t memAddress, ui
-    rtn->accelGyroMagnetismData_ ->magnetoZ_ = dataRead[32];
-
-    rtn->barometerData_ ->pressure_ = dataRead[36];
-    rtn->barometerData_ ->temperature_ = dataRead[40];
-
-    rtn->combustionChamberPressureData_ ->pressure_ = dataRead[44];
-
-    rtn->gpsData_ ->time_ = dataRead[48];
-    rtn->gpsData_->latitude_.degrees_ = dataRead[52];
-    rtn->gpsData_->latitude_.minutes_ = dataRead[56];
-    rtn->gpsData_->longitude_.degrees_ = dataRead[60];
-    rtn->gpsData_->longitude_.minutes_ = dataRead[64];
-
-    rtn->gpsData_->antennaAltitude_.altitude_ = dataRead[68];
-    rtn->gpsData_->geoidAltitude_.altitude_ = dataRead[72];
-    rtn->gpsData_->totalAltitude_.altitude_ = dataRead[76];
-
-    rtn->oxidizerTankPressureData_ ->pressure_ = dataRead[80];
-
+    checkEEPROMBlocking();
+    readFromEEPROM(dataRead, sizeof(dataRead), memAddress);
+    //TODO: memAdress may not be corrrect bc it's never incremented
+
+    theLogEntry.accelX = dataRead[0];
+    theLogEntry.accelY = dataRead[4];
+    theLogEntry.accelY = dataRead[8];
+    theLogEntry.gyroX = dataRead[12];
+    theLogEntry.gyroY = dataRead[16];
+    theLogEntry.gyroZ = dataRead[20];
+    theLogEntry.magnetoX = dataRead[24];
+    theLogEntry.magnetoY = dataRead[28];
+    theLogEntry.magnetoZ = dataRead[32];
+    theLogEntry.barometerPressure = dataRead[36];
+    theLogEntry.barometerTemperature = dataRead[40];
+    theLogEntry.combustionChamberPressure = dataRead[44];
+    theLogEntry.oxidizerTankPressure = dataRead[48];
+    theLogEntry.gps_time = dataRead[52];
+    theLogEntry.latitude_degrees = dataRead[56];
+    theLogEntry.latitude_minutes = dataRead[60];
+    theLogEntry.longitude_degrees = dataRead[64];
+    theLogEntry.longitude_minutes = dataRead[68];
+    theLogEntry.altitude = dataRead[72];
+    theLogEntry.currentFlightPhase = dataRead][76];
+    theLogEntry.tick = dataRead[80];
 }
 
-
-void buildLogEntry(AllData* data, char* buffer)
+/**
+ * @brief this method initializes the log entry struct
+ * and is only called during the start of the task
+ */
+void initializeLogEntry()
+{
+    theLogEntry.accelX = -1;
+    theLogEntry.accelY = -1;
+    theLogEntry.accelZ = -1;
+    theLogEntry.gyroX = -1;
+    theLogEntry.gyroY = -1;
+    theLogEntry.gyroZ = -1;
+    theLogEntry.magnetoX = -1;
+    theLogEntry.magnetoY = -1;
+    theLogEntry.magnetoZ = -1;
+    theLogEntry.barometerPressure = -1;
+    theLogEntry.barometerTemperature = -1;
+    theLogEntry.combustionChamberPressure = -1;
+    theLogEntry.oxidizerTankPressure = -1;
+    theLogEntry.gps_time = 0xFFFF;
+    theLogEntry.latitude_degrees = -1;
+    theLogEntry.latitude_minutes = 0xFFFF;
+    theLogEntry.longitude_degrees = -1;
+    theLogEntry.longitude_minutes = 0xFFFF;
+    theLogEntry.antennaAltitude = -1;
+    theLogEntry.geoidAltitude = -1;
+    theLogEntry.altitude = -1;
+    theLogEntry.currentFlightPhase = getCurrentFlightPhase();
+    theLogEntry.tick = HAL_GetTick();
```

Comment:
Could probably replace this with a single `memcpy(...)` (See http://www.cplusplus.com/reference/cstring/memcpy/ if you're unfamiliar with it) setting the entire entry to -1 (0xFF in every byte), followed by the two last lines that set actual values.

i.e.
```c
    memcpy(/* I'll let you figure out the args :) */);
    theLogEntry.currentFlightPhase = getCurrentFlightPhase();
    theLogEntry.tick = osKernelSysTick();
```